src/database.c

```diff
@@ -246,14 +246,16 @@ log_database_get_previous_chat(const gchar* const contact_barejid, char* start_t
 
     // Flip order when querying older pages
     gchar* sort = !flip ? "ASC" : "DESC";
-    gchar* end_date_fmt = end_time ? end_time : g_date_time_format_iso8601(g_date_time_new_now_local());
+    GDateTime* now = g_date_time_new_now_local();
+    gchar* end_date_fmt = end_time ? end_time : g_date_time_format_iso8601(now);
     gchar* start_date_fmt = start_time ? start_time : NULL;
     query = sqlite3_mprintf("SELECT * FROM (SELECT `message`, `timestamp`, `from_jid`, `type` from `ChatLogs` WHERE ((`from_jid` = '%q' AND `to_jid` = '%q') OR (`from_jid` = '%q' AND `to_jid` = '%q')) AND `timestamp` < '%q' AND (%Q IS NULL OR `timestamp` > %Q) ORDER BY `timestamp` DESC LIMIT %d) ORDER BY `timestamp` %s;", contact_barejid, myjid->barejid, myjid->barejid, contact_barejid, end_date_fmt, start_date_fmt, start_date_fmt, MESSAGES_TO_RETRIEVE, sort);
     if (!query) {
         log_error("log_database_get_previous_chat(): SQL query. could not allocate memory");
         return NULL;
     }
 
+    g_date_time_unref(now);
```

Comment:
I think you can free this directly after `g_date_time_format_iso8601()`. Or at least before the `if !(query)` otherwise we leak this in the error case. (Though that error case is about allocating memory and should almost never happen..)